.github/workflows/full-site-editing-plugin.yml

```diff
@@ -27,16 +25,87 @@ jobs:
         key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
 
     - name: Install dependencies
-      run: yarn install --frozen-lockfile
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: yarn install --frozen-lockfile # Not needed when restoring from cache.
+
+    - name: Build packages
+      if: steps.cache.outputs.cache-hit == 'true'
+      run: yarn run postinstall # Needed only when not running yarn install.
 
     - name: Build FSE
-      run: cd apps/full-site-editing && yarn build
+      run: yarn build
+      working-directory: apps/full-site-editing
 
     - name: Upload build artifact
       uses: actions/upload-artifact@v1
       with:
         name: fse-build-archive
         path: apps/full-site-editing/full-site-editing-plugin
 
+  mc_upload:
+    name: Create wpcom sync diff
+    needs: build
+    runs-on: ubuntu-latest
+    env:
+      CALYPSO_APP: fse_plugin
+      TRIGGER_CALYPSO_APP_BUILD_ENDPOINT: ${{ secrets.TRIGGER_CALYPSO_APP_BUILD_ENDPOINT }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@master
+
+    # It saves a bit of time to download the artifact rather than doing a build.
+    - name: Get Build
+      uses: actions/download-artifact@master
+      with:
+        name: fse-build-archive
+        path: apps/full-site-editing/full-site-editing-plugin
+
     - name: Send hook to Mission Control
       run: .github/workflows/send-calypso-app-build-trigger.sh
+
+  phpunit:
+    name: Run phpunit tests
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@master
+
+    # It saves a bit of time to download the artifact rather than doing a build.
+    - name: Get build
+      uses: actions/download-artifact@master
+      with:
+        name: fse-build-archive
+        path: apps/full-site-editing/full-site-editing-plugin
+
+    - name: Run composer
+      uses: nick-zh/composer-php@master
+      with:
+        action: 'install'
+
+    # We still need to access some local node modules to run things.
+    - name: Restore node_modules cache
+      id: cache
+      uses: actions/cache@master
+      with:
+        path: |
+          node_modules
+          */*/node_modules
+        key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+
+    - name: Install dependencies
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: yarn install --frozen-lockfile
+
+    - name: Setup wp-env dependencies
+      run: |
+        echo '{ "plugins": [ "./dev-plugin", "./full-site-editing-plugin", "https://downloads.wordpress.org/plugin/gutenberg.latest-stable.zip" ], "themes": [] }' > .wp-env.override.json
+      working-directory: apps/full-site-editing
+
+    - name: Start wp-env
+      run: npx wp-env start || npx wp-env start # Retry if failed.
```

Comment:
In an ideal world, we would configure the `container` part of of this job to be a WordPress instance instead of starting another one inside of it. That would be quite a bit faster. But wp-env provides a lot of stuff for us here:

1. Automatically installs the Gutenberg plugin
2. Automatically installs WordPress
3. Automatically provides a wp config that works with phpunit
4. Provides a service which configures everything correctly for phpunit.